### ISSUE TITLE
Monaco Editor integration for Web UI

### ISSUE THREAD

**Issue by xerial** [wvlet/wvlet#81]
Implement wvlet editor in wvlet-ui module
- Example code for using Monaco editor in Scala.js https://github.com/wvlet/querybase-legacy/blob/master/querybase-ui/src/main/scala/wvlet/querybase/ui/component/editor/TextEditor.scala
- Add syntax highlighter for Monaco https://microsoft.github.io/monaco-editor/monarch.html

**Comment by xerial** [comment]
Related items: 
- #302
- #303 
- #300

**Comment by Matts966** [comment]
<img width="638" alt="Screenshot 2024-10-16 at 15 11 26" src="https://github.com/user-attachments/assets/66bc2614-4700-4701-bbe2-49e9efbab31a">
Diagnostics example

```scala
import typings.monacoEditor.mod.MarkerSeverity
import typings.monacoEditor.mod.editor.ITextModel

  override def onMount: Unit =
    val instance = editor.create(
      dom.document.getElementById("editor").asInstanceOf[dom.HTMLElement],
      monacoEditorOptions
    )
    // val model = editor.createModel("test")
    editor.setModelMarkers(
      instance.getModel().asInstanceOf[ITextModel], "test", js.Array(
      new:
        val startLineNumber = 1
        val startColumn = 2
        val endLineNumber = 1
        val endColumn = 4
        val message = "ok"
        val severity = MarkerSeverity.Error
    ));
```

**Comment by xerial** [comment]
## Building Web UI with Monaco editor
- editor (feature): Add Web UI template [#299](https://github.com/wvlet/wvlet/pull/299)
- editor (feature): Add Monaco editor to UI [#300](https://github.com/wvlet/wvlet/pull/300)
- editor (feature): Add monaco highlight [#303](https://github.com/wvlet/wvlet/pull/303) 
- editor (feature): Support multiline comment and interpolation [#308](https://github.com/wvlet/wvlet/pull/308)
- editor (feature): Add keyword completion [#309](https://github.com/wvlet/wvlet/pull/309) 
- editor (fix): Fix monaco rendering error [#310](https://github.com/wvlet/wvlet/pull/310)
- editor (feature): Fill the entire screen [#311](https://github.com/wvlet/wvlet/pull/311)
- editor (feature): Continuously fetch QueryInfo and show logs to console window [#312](https://github.com/wvlet/wvlet/pull/312)
- editor (feature): Show result preview [#313](https://github.com/wvlet/wvlet/pull/313)
- editor (feature): Show query error message in the console [#314](https://github.com/wvlet/wvlet/pull/314)
- editor (feature): Use fixed editor layout [#317](https://github.com/wvlet/wvlet/pull/317)
- editor (feature): Stabilize editor height [#318](https://github.com/wvlet/wvlet/pull/318)
- editor (feature): Support test/describe with ctrl-j ctrl-t/ctrl-d shortcut keys [#319](https://github.com/wvlet/wvlet/pull/319)

## wvlet ui command 
- #322  
- #324 
- #325
- #327 
## Wvlet query runner
- editor (feature): Add RPC interface [#302](https://github.com/wvlet/wvlet/pull/302) 
- editor (feature): Integrate UI with RPC server [#306](https://github.com/wvlet/wvlet/pull/306) 
- editor (feature): Run query from Web UI [#307](https://github.com/wvlet/wvlet/pull/307)

**Comment by xerial** [comment]
Now, the initial version of the Wvlet Monaco editor can be started with the `wvlet ui` command. More improvements will be necessary for text completion and executor, which will be addressed in different milestones.
